File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int L = 0;
        int min = Integer.MAX_VALUE;
        int sum = 0;

        for (int R = 0; R < nums.length; R++) {
            sum += nums[R];

            while (sum >= target) {
                min = Math.min(min, R - L + 1);
                sum -= nums[L];
                L++;
            }
        }
        return min == Integer.MAX_VALUE ? 0 : min;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        L = 0
        R = len(nums) - 1

        while L < R:
            total = nums[L] + nums[R]

            if total == target:
                return [L + 1, R + 1]

            if total < target:
                L += 1
            else:
                R -= 1

        return [-1, -1]
